allow anonymous users to access @types endpoint

erral opened the pull request:
Addresses #1409

mister-roboto commented:
@erral thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

tisto commented:
@erral what's the status of this PR? Can you elaborate your use case?

erral commented:
I don't recall why we added this PR...

I think it must be somehow related to the work we did during the Beethoven Sprint when we added in Volto a default view for schema based content-types

davisagli commented:
@tisto @erral I'm aware of 2 use cases that came up for this.

The first is the default schema-based view, like you said. That still does not work for anonymous users. But, this change stalled because there were some concerns about exposing private values in the schemas (such as vocabulary values).

The second is the use case of showing content type titles in listings. We solved that a different way in https://github.com/plone/plone.restapi/pull/1355
